core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -448,6 +448,31 @@ public void unparseCall(SqlWriter writer, SqlCall call, int leftPrec,
     call.getOperator().unparse(writer, call, leftPrec, rightPrec);
   }
 
+  public void unparseRow(SqlWriter writer,
+      SqlCall call, int leftPrec,  int rightPrec) {
+    if (getConformance().allowExplicitRowValueConstructor()) {
+      SqlUtil.unparseFunctionSyntax(call.getOperator(), writer, call);
+    } else {
+      if (" ".equals(call.getOperator().getName())) {
+        writer.print(call.getOperator().getName());
+      }
+      if (writer.isAlwaysUseParentheses()) {
+        for (SqlNode operand : call.getOperandList()) {
+          writer.sep(",");
```

Comment:
Can you explain why we need this decision ?

Reply:
This `isAlwaysUseParentheses` parameter will link operations with parentheses.
For example ,
`(a + b) * c`.The fully-parenthesized expression, `((a + b) * c)`

If this parameter is in the `SqlRowOperator`. There will be extra parentheses
For example,
`isAlwaysUseParentheses=true` and `allowExplicitRowValueConstructor=false`

input expression:
`insert into emp valuse (1,'a')`

unparse result:
`insert into emp valuse ((1,'a'))`

I think this is a wrong SQL.